lib/galaxy/authnz/custos_authnz.py

```diff
@@ -230,3 +247,12 @@ def _get_verify_param(self):
             return self.config['ca_bundle']
         else:
             return self.config['verify_ssl']
+
+    def generate_username(self, trans, email):
+        temp_username = email.split('@')[0] #username created from username portion of email
+        count = 0
+
+        while (trans.sa_session.query(trans.app.model.User).filter_by(username=(temp_username+str(count))).first()):
```

Comment:
this doesn't check for existing username without the appended count - i.e., you start with appending zero (i.e., before probing for `jack0`, you need to check the availability of `jack`, right?)